src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -15,22 +15,25 @@ namespace Microsoft.CSharp.RuntimeBinder
 {
     internal static class BinderHelper
     {
+        private static MethodInfo s_DoubleIsNaN;
+        private static MethodInfo s_SingleIsNaN;
+
         internal static DynamicMetaObject Bind(
                 DynamicMetaObjectBinder action,
                 RuntimeBinder binder,
-                IEnumerable<DynamicMetaObject> args,
+                DynamicMetaObject[] args,
                 IEnumerable<CSharpArgumentInfo> arginfos,
                 DynamicMetaObject onBindingError)
         {
-            List<Expression> parameters = new List<Expression>();
+            Expression[] parameters = new Expression[args.Length];
             BindingRestrictions restrictions = BindingRestrictions.Empty;
             ICSharpInvokeOrInvokeMemberBinder callPayload = action as ICSharpInvokeOrInvokeMemberBinder;
             ParameterExpression tempForIncrement = null;
             IEnumerator<CSharpArgumentInfo> arginfosEnum = arginfos == null ? null : arginfos.GetEnumerator();
 
-            int index = 0;
-            foreach (DynamicMetaObject o in args)
+            for (int index = 0; index != args.Length; ++index)
```

Comment:
Hmm. I suppose it would be extra work to be sure `index` wasn't changed elsewhere. What a pity I got into the habit of using `!=` from STL iterators that would more often have `!=` and no `<` than the other way around, especially since I'm so rusty at that that the habit has no actual application.

Does postfix vs prefix matter (no strong counter-preference there, just curious).